tests/compiler/test_loops.py

```diff
@@ -173,6 +173,14 @@ def test_resultcollector_receive_result_flatten(snippetcompiler) -> None:
                     assert.success = false
                 end
             end
+
```

Comment:
@sanderr this testcase alerted me to some existing bugs in the execute_direct and as_constant